src/core/annotation.js

```diff
@@ -670,6 +692,65 @@ class Annotation {
       stream.reset();
     }
   }
+
+  /**
+   * Construct the (fully qualified) field name from the (partial) field
+   * names of the field and its ancestors.
+   *
+   * @private
+   * @memberof WidgetAnnotation
+   * @param {Dict} dict - Complete widget annotation dictionary
+   * @returns {string}
+   */
+  _constructFieldName(dict) {
+    // Both the `Parent` and `T` fields are optional. While at least one of
+    // them should be provided, bad PDF generators may fail to do so.
+    if (!dict.has("T") && !dict.has("Parent")) {
+      warn("Unknown field name, falling back to empty field name.");
```

Comment:
*Connect the the comments above:* By always running this method, for every Annotation rather than just WidgetAnnotations, won't this warning now be printed *a lot* for cases where it shouldn't?
Obviously we could remove it and/or change it to `info` instead, but I do wonder about these changes generally.